bigquery_etl/cli/view.py

```diff
@@ -378,13 +381,32 @@ def clean(
         )
 
 
-def _list_managed_views(client, dataset, pattern):
+def _list_managed_views(client, dataset, pattern, skip_authorized):
+    query = f"""
+      SELECT
+        table_catalog || "." || table_schema || "." || table_name AS table_id,
+        CONTAINS_SUBSTR(option_value, 'STRUCT("authorized", "")') AS is_authorized
+      FROM
+        `{dataset.dataset_id}.INFORMATION_SCHEMA.VIEWS`
+      INNER JOIN
+        `{dataset.dataset_id}.INFORMATION_SCHEMA.TABLE_OPTIONS`
+      USING
+        (table_catalog,
+         table_schema,
+         table_name)
+      WHERE
+        option_name = "labels"
+      AND CONTAINS_SUBSTR(option_value, 'STRUCT("managed", "")')
+    """
+
+    # running a query against information schema instead of using the API to list tables is much faster
+    job = client.query(query)
+    result = list(job.result())
     return [
-        table
-        for table in client.list_tables(dataset)
-        if table.table_type == "VIEW"
-        and "managed" in table.labels
-        and (pattern is None or fnmatchcase(sql_table_id(table), f"*{pattern}"))
+        row.table_id
+        for row in result
+        if (pattern is None or fnmatchcase(sql_table_id(row.table_id), f"*{pattern}"))
+        and (not skip_authorized or not row.is_authorized)
```

Comment:
moving `skip_authorized` into this function, so that we can just return the view IDs